src/parser/CodeBlock.h

```diff
@@ -63,34 +63,159 @@ struct NativeFunctionInfo {
 
 class CallNativeFunctionData : public gc {
 public:
+    CallNativeFunctionData(NativeFunctionPointer fn)
+        : m_fn(fn)
+    {
+    }
+
     NativeFunctionPointer m_fn;
 };
 
+class NativeCodeBlock;
 class InterpretedCodeBlock;
 
 class CodeBlock : public gc {
+public:
+    static const unsigned MAX_FUNCTION_LENGTH = 65536;
+
+    void* operator new(size_t size) = delete;
+    void* operator new[](size_t size) = delete;
+
+    virtual uint16_t functionLength() const = 0;
+    virtual AtomicString functionName() const = 0;
+
+    virtual bool isNativeCodeBlock() const
+    {
+        return false;
+    }
+
+    virtual bool isInterpretedCodeBlock() const
+    {
+        return false;
+    }
+
+    NativeCodeBlock* asNativeCodeBlock()
+    {
+        ASSERT(isNativeCodeBlock());
+        return (NativeCodeBlock*)this;
+    }
+
+    InterpretedCodeBlock* asInterpretedCodeBlock()
+    {
+        ASSERT(isInterpretedCodeBlock());
+        return (InterpretedCodeBlock*)this;
+    }
+
+    Context* context()
+    {
+        return m_context;
+    }
+
+    bool isStrict() const
+    {
+        return m_isStrict;
+    }
+
+protected:
+    CodeBlock()
+        : m_context(nullptr)
+    {
+    }
+
+    CodeBlock(Context* ctx, bool isStrict)
+        : m_context(ctx)
+        , m_isStrict(isStrict)
+    {
+    }
+
+    Context* m_context;
+    bool m_isStrict;
```

Comment:
How about to remove this variable if possible?
4bytes for just 1 flag is not looks good

Reply:
Thanks for the review. This boolean member is moved into NativeCodeBlock and InterpretedCodeBlock each.